ignite/pkg/jsonfile/jsonfile.go

```diff
@@ -0,0 +1,327 @@
+package jsonfile
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/pkg/errors"
+
+	"github.com/ignite/cli/ignite/pkg/tarball"
+)
+
+const (
+	keySeparator = "."
+)
+
+var (
+	// ErrFieldNotFound parameter not found into json
+	ErrFieldNotFound = errors.New("JSON field not found")
+
+	// ErrInvalidValueType invalid value type
+	ErrInvalidValueType = errors.New("invalid value type")
+
+	// ErrInvalidURL invalid file URL
+	ErrInvalidURL = errors.New("invalid file URL")
+)
+
+type (
+	// JSONFile represents the JSON file and also implements the io.write interface
+	// saving directly to the file
+	JSONFile struct {
+		file        ReadWriteSeeker
+		tarballPath string
+		url         string
+		updates     map[string][]byte
+		cache       []byte
+	}
+
+	// UpdateFileOption configures file update function with key and value
+	UpdateFileOption func(map[string][]byte)
+)
+
+type (
+	// writeTruncate represents the truncate method from io.WriteSeeker interface
+	writeTruncate interface {
+		Truncate(size int64) error
+	}
+
+	// ReadWriteSeeker represents the owns ReadWriteSeeker interface inherit from io.ReadWriteSeeker
+	ReadWriteSeeker interface {
+		io.ReadWriteSeeker
+		Close() error
+		Sync() error
+	}
+)
+
+// New creates a new JSONFile
+func New(file ReadWriteSeeker) *JSONFile {
+	return &JSONFile{
+		updates: make(map[string][]byte),
+		file:    file,
+	}
+}
+
+// FromPath parse JSONFile object from path
+func FromPath(path string) (*JSONFile, error) {
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_APPEND, 0o600)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot open the file")
+	}
+	return New(file), nil
+}
+
+// FromURL fetches the file from the given URL and returns its content.
+func FromURL(ctx context.Context, url, destPath, tarballFileName string) (*JSONFile, error) {
```

Comment:
```suggestion
// FromURL fetches the file from the given URL and returns its content.
// If tarballFileName is not empty, the URL is interpreted as a tarball file,
// tarballFileName is extracted from it and is returned instead of the URL
// content.
func FromURL(ctx context.Context, url, destPath, tarballFileName string) (*JSONFile, error) {
```

Took me a while to understand that by reading the code, so a comment is maybe not too much :)